rpc/lib/server/handlers.go

```diff
@@ -228,25 +228,25 @@ func makeHTTPHandler(rpcFunc *RPCFunc, logger log.Logger) func(http.ResponseWrit
 	// Exception for websocket endpoints
 	if rpcFunc.ws {
 		return func(w http.ResponseWriter, r *http.Request) {
-			WriteRPCResponseHTTPError(w, http.StatusMethodNotAllowed, types.NewRPCResponse("", nil, "This RPC method is only for websockets"))
+			WriteRPCResponseHTTP(w, types.RPCInternalError("", errors.New("Trying to use Websocket method in non-ws context")))
```

Comment:
should we be doing JSONRPC error codes for the makeHTTPHandler? yes the response is JSON, but since the request wasn't done via JSONRPC, perhaps we should be returning http error codes ? ie. we only follow the JSONRPC spec when the request is via JSONRPC

Reply:
> but since the request wasn't done via JSONRPC

yes, but the format in use is JSONRPC (i.e. both request and response objects contain `jsonrpc` field). I think it is ok to leave the error codes

Reply:
Ok. Just to re-iterate, it means it will return 200, and the caller has to check the json for the error. I guess it's fine